app/assets/javascripts/client/consensus/ConsensusService.js

```diff
@@ -0,0 +1,153 @@
+(function() {
+  'use strict';
+  angular.module('PDRClient')
+      .service('ConsensusService', ConsensusService);
+
+  ConsensusService.$inject = [
+    '$stateParams',
+    '$location',
+    '$q',
+    'Consensus',
+    'AnalysisConsensus',
+    'AnalysisResponse',
+    'Score',
+    'AnalysisConsensusScore',
+    'ProductEntry',
+    'DataEntry'
+  ];
+
+  function ConsensusService($stateParams, $location, $q, Consensus, AnalysisConsensus, AnalysisResponse, Score, AnalysisConsensusScore, ProductEntry, DataEntry) {
+    var service = this;
+
+    service.setContext = function(context) {
+      service.context = context;
+      return service;
+    };
+
+    service.extractId = function() {
```

Comment:
I've seen this in a couple of places now, the name `extractId` doesn't really tell me what `ID` is being extracted, is it the `currentId` or the `parentId`?